### DISCUSSION TITLE
FP16 does not decrease much GPU memory

### DISCUSSION THREAD

**Discussion by icoz69** [Lightning-AI/pytorch-lightning#13252] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
hi, I just tried mix precision training with precision=16 set in the trainer. I found the training speed does increase by around 30%, but the GPU memory merely decreases. Should it be half of the raw memory?

**Comment by akihironitta** [comment, marked as answer]
Hi @icoz69 AFAIK, the memory usage depends on your model architecture, specifically, the ratio of the model size to the size of activations. This is because, with amp, your model always stays in fp32 while some operations in your model are done in fp16.

**Reply by m13uz** [reply]
which operations specifically are done in fp16?

**Reply by akihironitta** [reply]
See https://pytorch.org/docs/stable/amp.html#autocast-op-reference.
